plugins/newsql/newsql.c

```diff
@@ -2025,18 +2025,22 @@ void free_newsql_appdata(struct sqlclntstate *clnt)
     free(appdata->col_info.type);
 }
 
-extern void * (*externGetAuthData) (void*);
-extern void * (*externalMakeAuthData) (const char *, int, int, int, void *);
+extern void * (*externalMakeAuthData) (void*, const char *, int, int, int, void *);
 
-void * getAuthData(void * p_clnt) {
-    struct sqlclntstate *clnt = p_clnt;
+static void * newsql_get_authdata(struct sqlclntstate *clnt)
+{
     struct newsql_appdata *appdata = clnt->appdata;
     if (appdata) {
         CDB2SQLQUERY *sql_query = appdata->sqlquery;
-        if (sql_query && sql_query->identity) {
-            return externalMakeAuthData(sql_query->identity->principal, sql_query->identity->majorversion, sql_query->identity->minorversion, sql_query->identity->data.len, sql_query->identity->data.data);
+        if (sql_query && sql_query->identity
+            && sql_query->identity->principal && sql_query->identity->data.data) {
+            return externalMakeAuthData(clnt->authdata, sql_query->identity->principal, sql_query->identity->majorversion, sql_query->identity->minorversion, sql_query->identity->data.len, sql_query->identity->data.data);
```

Comment:
Is it possible to pass `sql_query->identity` instead?

Reply:
ok

Reply:
Having auth plugin independent of cdb2api (protobuf) will be better

Reply:
Does it make sense to expose `struct authenticationData` in `sql.h` so all types of requests know what it is and can pass it in?

Reply:
That's part auth plugin, its individual components should not be used outside of that plugin.

Reply:
All components need to be accessible to anyone who wants to use it. Instead of passing each individual component, it would be cleaner to pass a struct ref. This doesn't really expose any code from the plugin. I'm just proposing that instead of `externalMakeAuthData(6 arguments)` we make the signature `externalMakeAuthData(authenticationData *)`

Reply:
Or perhaps `db_access.h` is more suitable to have that struct. Also move:
`extern void *(*externalMakeAuthData)(void *, const char *, int, int, int, void *);` to the same header (which ever you chose sql or db_access)

Reply:
The function gets the required data to make that structure only
https://bbgithub.dev.bloomberg.com/comdb2/bb-plugins/blob/master/auth/auth.cpp#L162
https://bbgithub.dev.bloomberg.com/comdb2/bb-plugins/blob/master/auth/auth.cpp#L82

Reply:
passed identity and made it a separate function specific to newsql